reference/configuration/framework.rst

```diff
@@ -169,39 +293,431 @@ see :doc:`/cookbook/request/load_balancer_reverse_proxy`.
+
+            <framework:config>
+                <esi />
+            </framework:config>
+        </container>
+
+    .. code-block:: php
+
+        // app/config/config.php
+        $container->loadFromExtension('framework', array(
+            'esi' => true,
+        ));
+
+fragments
+~~~~~~~~~
+
+.. seealso::
+
+    Learn more about fragments in the
+    :ref:`HTTP Cache article <book-http_cache-fragments>`.
+
+.. _reference-fragments-enabled:
+
+enabled
+.......
+
+**type**: ``boolean`` **default**: ``false``
+
+Whether to enable the fragment listener or not. The fragment listener is used
+to render ESI fragments independently of the rest of the page.
+
+This setting is automatically set to ``true`` when one of the child settings is
+configured.
+
+.. _reference-fragments-path:
+
+path
+....
+
+**type**: ``string`` **default**: ``'/_fragment'``
+
+The path prefix for fragments. The fragment listener will only be executed
+when the request starts with this path. It becomes the service container
+parameter called ``fragment.path``.
+
+profiler
+~~~~~~~~
+
+.. _reference-profiler-enabled:
+
+enabled
+.......
+
+.. versionadded:: 2.2
+    The ``enabled`` option was introduced in Symfony 2.2. Previously, the profiler
+    could only be disabled by omitting the ``framework.profiler`` configuration
+    entirely.
+
+**type**: ``boolean`` **default**: ``true`` in the ``dev`` and ``test`` environments
+
+The profiler can be disabled by setting this option to ``false``.
+
+.. note::
+
+    The profiler works independently from the Web Developer Toolbar, see the
+    :doc:`WebProfilerBundle configuration
+    </reference/configuration/web_profiler>` on how to disable/enable it.
+
+collect
+.......
+
+.. versionadded:: 2.3
+    The ``collect`` option was introduced in Symfony 2.3. Previously, when
+    ``profiler.enabled`` was ``false``, the profiler *was* actually enabled,
+    but the collectors were disabled. Now, the profiler and the collectors
+    can be controlled independently.
+
+**type**: ``boolean`` **default**: ``true``
+
+This option configures the way the profiler behaves when it is enabled. If set
+to ``true``, the profiler collects data for all requests. If you want to only
+collect information on-demand, you can set the ``collect`` flag to ``false``
+and activate the data collectors manually::
+
+    $profiler->enable();
+
+only_exceptions
+...............
+
+**type**: ``boolean`` **default**: ``false``
+
+When this is set to ``true``, the profiler will only be enabled when an
+exception is thrown during the handling of the request. It becomes the service
+container parameter called ``profiler_listener.only_exceptions``.
+
+only_master_requests
+....................
+
+**type**: ``boolean`` **default**: ``false``
+
+When this is set to ``true``, the profiler will only be enabled on the master
+requests (and not on the subrequests). It becomes the service container
+parameter called ``profiler_listener.only_master_requests``.
+
+dsn
```

Comment:
We actually have a cookbook entry on this! http://symfony.com/doc/current/cookbook/profiler/storage.html

Perhaps we can unify them a bit - that is a pretty short entry, and it's not great. You listing the valid DSN's below is better than what it does (listing the internal classes used does not seem helpful).